src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingLeadingAsteriskCheck.java

```diff
@@ -0,0 +1,108 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * <p>
+ * Checks if the javadoc has leading asterisk on each line or not.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingLeadingAsterisk"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * &#47;** // violation, javadoc has lines without leading asterisk.
+ *  * Some description here.
+ *    Another line of description.
+ *  *&#47;
+ * public class Test {
+ *   &#47;** // violation, javadoc has lines without leading asterisk.
+ *    * Some description here.
+ *    * Another line of description.
+ *      Another line of description.
+ *    *&#47;
+ *   public void test() {}
+ *
+ *   &#47;** Some description here. *&#47; // OK
+ *   public void test1() {}
+ * }
+ * </pre>
+ *
+ * @since 8.32
+ */
+@StatelessCheck
+public class MissingLeadingAsteriskCheck extends AbstractCheck {
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_MISSING_ASTERISK = "javadoc.missing.asterisk";
+
+    /**
+     * The new line character that separates each line of javadoc.
+     */
+    private static final String COMMENT_SEPARATOR = "\n";
```

Comment:
I can iterate through each `AST` child instead until I find `BLOCK_COMMMENT_END`. Is this okay?

Reply:
@strkkk I just checked it out. We can't iterate through the AST nodes. Because the whole comment content is one AST node. Can you suggest me some other way?

Reply:
You can take line number of block begin and block end, after it get file content https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/FileContents.java
and iterate over lines there.
Usage of this class is not encouraged, but as for me it looks better than possible problems with line separators

Reply:
I tried this method. But the `getLine()` method in `FileContents` returns the following String
` * // violation\n  ** OK\n *\n  *\n violation`. It is wrapping all the comment lines into one single string.

Reply:
Ok. But this line separator can be a problem. Can you try with CR line separator? I am afraid it wont report violation in this case.

Reply:
Sure. Let me add more cases to the input.

Reply:
@strkkk I have an idea. The different line separators LN, CR, CRLN are represented as `\n`, `\r`, `\n\r` in `String`. So what if we write a Regex and pass this regex to `split` function of `String`.

Reply:
We can try this

Reply:
I have updated the code. Added the regex here https://github.com/checkstyle/checkstyle/pull/8085/files#diff-1c1129c743f863d78fcd7fba8d0a81e9R73
Please have a look